src/interpreter/bytecode-array-builder.cc

```diff
@@ -1394,13 +1394,27 @@ BytecodeArrayBuilder& BytecodeArrayBuilder::RecordReplayAssertValue(const std::s
   return *this;
 }
 
-BytecodeArrayBuilder& BytecodeArrayBuilder::RecordReplayInstrumentation(const char* kind,
-                                                                        int source_position) {
+int BytecodeArrayBuilder::RecordReplayRegisterInstrumentationSite(
+    const char* kind, int source_position) {
+  if (record_replay_instrumentation_site_locations_.find(source_position) !=
+      record_replay_instrumentation_site_locations_.end()) {
+    // Don't insert a site at the same location more than once.
+    return -1;
```

Comment:
This filtering should only be done for breakpoint sites.  The other instrumentation kinds need to be emitted regardless of whether the source position has previously been used with another instrumentation site, or we can end up with broken behavior where e.g. we enter frames but don't exit them later.

It would also be good to only filter if the source position is not kNoSourcePosition.  I don't know if that ever affects breakpoints but this logic doesn't really make sense when there's no source position.